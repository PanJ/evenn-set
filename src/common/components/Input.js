import React from 'react'

const Input = ({
  value,
  onChange,
  placeholder,
  error,
  icon,
  label,
  type = 'text'
}) => (
  <div className="field">
    <label className="label">{label}</label>
    <div
      className={`control ${icon ? 'has-icons-left' : ''} ${
        error ? 'has-icons-right' : ''
      }`}
    >
      <input
        value={value}
        onChange={e => onChange(e.target.value)}
        className={`input ${error ? 'is-danger' : ''}`}
        type={type}
        placeholder={placeholder}
      />
      {icon && (
        <span className="icon is-small is-left">
          <i className={`fas fa-${icon}`} />
        </span>
      )}

      {error && (
        <span className="icon is-small is-right">
          <i className="fas fa-exclamation-triangle" />
        </span>
      )}
    </div>
    {error && <p className="help is-danger">{error}</p>}
  </div>
)

export default Input
