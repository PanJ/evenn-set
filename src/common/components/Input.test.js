import React from 'react'
import ReactDOM from 'react-dom'
import Input from './Input'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const emailElement = (
  <Input label="Email" type="email" placeholder="Email input" />
)

describe('<Input />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Input />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('matches snapshot', () => {
    const component = renderer.create(emailElement)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should correctly render label', () => {
    const wrapper = shallow(emailElement)
    expect(wrapper.find('label').text()).toEqual('Email')
  })
  it('should call onChange', () => {
    const mockOnChange = jest.fn()
    const wrapper = shallow(
      <Input
        label="Email"
        type="email"
        value=""
        onChange={mockOnChange}
        placeholder="Email input"
      />
    )
    wrapper.find('input').simulate('change', { target: { value: 'a' } })
    expect(mockOnChange.mock.calls.length).toEqual(1)
    expect(mockOnChange.mock.calls[0][0]).toEqual('a')
  })
})
