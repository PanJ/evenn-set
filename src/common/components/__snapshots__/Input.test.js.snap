// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<div
  className="field"
>
  <label
    className="label"
  >
    Email
  </label>
  <div
    className="control  "
  >
    <input
      className="input "
      onChange={[Function]}
      placeholder="Email input"
      type="email"
      value={undefined}
    />
  </div>
</div>
`;
